File: reset_all_timeframes.py
#!/usr/bin/env python3
"""
Script para desactivar TODAS las temporalidades de TODOS los activos
en la colección 'monitoreos' de Firestore.

Uso: python3 reset_all_timeframes.py
"""

import os
import sys
from pathlib import Path
from google.cloud import firestore
from google.oauth2 import service_account

# Configurar credenciales
SCRIPT_DIR = Path(__file__).parent
CREDENTIALS_PATH = SCRIPT_DIR / "trading-firestore.json"

if not CREDENTIALS_PATH.exists():
    print(f"❌ Credentials no encontrados: {CREDENTIALS_PATH}")
    sys.exit(1)

print("🔍 Conectando a Firestore...")
credentials = service_account.Credentials.from_service_account_file(str(CREDENTIALS_PATH))
db = firestore.Client(project='trading-449607', credentials=credentials)

def reset_all_timeframes():
    monitoreos_ref = db.collection('monitoreos')
    
    # Obtener todos los documentos de monitoreos
    print("📋 Obteniendo todos los documentos de monitoreos...")
    docs = monitoreos_ref.stream()
    
    doc_list = list(docs)
    if not doc_list:
        print("✅ No hay documentos en monitoreos. Nada que hacer.")
        return
    
    print(f"📊 Encontrados {len(doc_list)} documentos de monitoreos")
    
    updated_count = 0
    batch_size = 500
    batches = []
    current_batch = db.batch()
    batch_count = 0
    
    for doc in doc_list:
        data = doc.to_dict()
        symbol = data.get('symbol', 'unknown')
        exec_id = data.get('exec_id', 'unknown')
        
        # Resetear todas las temporalidades
        current_batch.update(doc.reference, {
            'running': [],                    # Sin TFs corriendo
            'estado': 'stopped',              # Estado detenido
            'updated_at': firestore.SERVER_TIMESTAMP,
        })
        
        batch_count += 1
        updated_count += 1
        print(f"  ✓ {symbol} ({exec_id}): timeframes desactivados")
        
        # Commit del batch cada 500 docs
        if batch_count >= batch_size:
            print(f"\n💾 Aplicando batch de {batch_count} cambios...")
            current_batch.commit()
            batches.append(batch_count)
            current_batch = db.batch()
            batch_count = 0
    
    # Commit final
    if batch_count > 0:
        print(f"\n💾 Aplicando último batch de {batch_count} cambios...")
        current_batch.commit()
        batches.append(batch_count)
    
    print(f"\n✅ ÉXITO: {updated_count} documentos actualizados en {len(batches)} batches")
    print("📝 Todas las temporalidades de todos los activos están ahora INACTIVAS")

if __name__ == "__main__":
    try:
        reset_all_timeframes()
        print("\n🎉 Proceso completado")
    except Exception as e:
        print(f"\n❌ ERROR: {e}")
        import traceback
        traceback.print_exc()
        sys.exit(1)

File: reset_all_timeframes.js
/**
 * Script para desactivar TODAS las temporalidades de TODOS los activos
 * en la colección 'monitoreos' de Firestore.
 * 
 * Uso: node reset_all_timeframes.js
 */

const { initializeApp, getApps, cert } = require('firebase-admin');
const fs = require('fs');
const path = require('path');

// Inicializar Firebase Admin
const credentialsPath = path.join(__dirname, 'trading-firestore.json');
if (!fs.existsSync(credentialsPath)) {
  console.error('❌ Credentials no encontrados:', credentialsPath);
  process.exit(1);
}

if (getApps().length === 0) {
  initializeApp({
    credential: cert(credentialsPath),
  });
}

const admin = require('firebase-admin');
const db = admin.firestore();

async function resetAllTimeframes() {
  console.log('🔍 Conectando a Firestore...');
  
  const monitoreosRef = db.collection('monitoreos');
  
  // Obtener todos los documentos de monitoreos
  console.log('📋 Obteniendo todos los documentos de monitoreos...');
  const snapshot = await monitoreosRef.get();
  
  if (snapshot.empty) {
    console.log('✅ No hay documentos en monitoreos. Nada que hacer.');
    return;
  }
  
  console.log(`📊 Encontrados ${snapshot.size} documentos de monitoreos`);
  
  const batch = db.batch();
  let updatedCount = 0;
  
  snapshot.forEach(doc => {
    const data = doc.data();
    const symbol = data.symbol || 'unknown';
    const execId = data.exec_id || 'unknown';
    
    // Resetear todas las temporalidades
    batch.update(doc.ref, {
      running: [],                    // Sin TFs corriendo
      selected_tfs: [],              // Sin TFs seleccionadas
      locked_timeframes: false,      // Desbloquear timeframes
      // Mantener allowed_timeframes si existe, pero vaciar selección
      updated_at: admin.firestore.FieldValue.serverTimestamp(),
    });
    
    updatedCount++;
    console.log(`  ✓ ${symbol} (${execId}): timeframes desactivados`);
  });
  
  // Commit del batch
  console.log('\n💾 Aplicando cambios...');
  await batch.commit();
  
  console.log(`\n✅ ÉXITO: ${updatedCount} documentos actualizados`);
  console.log('📝 Todas las temporalidades de todos los activos están ahora INACTIVAS');
}

// Ejecutar
resetAllTimeframes()
  .then(() => {
    console.log('\n🎉 Proceso completado');
    process.exit(0);
  })
  .catch(err => {
    console.error('\n❌ ERROR:', err);
    process.exit(1);
  });
